controllers/argocd/configmap_test.go

```diff
@@ -393,29 +393,65 @@ func TestReconcileArgoCD_reconcileArgoConfigMap_withDexConnector(t *testing.T) {
 			err := r.reconcileArgoConfigMap(a)
 			assert.NoError(t, err)
 
-			cm := &corev1.ConfigMap{}
-			err = r.Client.Get(context.TODO(), types.NamespacedName{
-				Name:      common.ArgoCDConfigMapName,
-				Namespace: testNamespace,
-			}, cm)
-			assert.NoError(t, err)
+			getDexConfig := func(t *testing.T, k8sClient client.Client) map[string]interface{} {
+				cm := &corev1.ConfigMap{}
+				err := k8sClient.Get(context.TODO(), types.NamespacedName{
+					Name:      common.ArgoCDConfigMapName,
+					Namespace: testNamespace,
+				}, cm)
+				assert.NoError(t, err)
+
+				dex, ok := cm.Data["dex.config"]
+				if !ok {
+					t.Fatal("reconcileArgoConfigMap with dex failed")
+				}
+
+				m := make(map[string]interface{})
+				err = yaml.Unmarshal([]byte(dex), &m)
+				assert.NoError(t, err, fmt.Sprintf("failed to unmarshal %s", dex))
 
-			dex, ok := cm.Data["dex.config"]
-			if !ok {
-				t.Fatal("reconcileArgoConfigMap with dex failed")
+				return m
 			}
 
-			m := make(map[string]interface{})
-			err = yaml.Unmarshal([]byte(dex), &m)
-			assert.NoError(t, err, fmt.Sprintf("failed to unmarshal %s", dex))
+			m := getDexConfig(t, r.Client)
+
+			verifyDexConnector := func(dexCfg map[string]interface{}) {
+				connectors, ok := dexCfg["connectors"]
+				if !ok {
+					t.Fatal("no connectors found in dex.config")
+				}
+				dexConnector := connectors.([]interface{})[0].(map[interface{}]interface{})
+				config := dexConnector["config"]
+				assert.Equal(t, config.(map[interface{}]interface{})["clientID"], "system:serviceaccount:argocd:argocd-argocd-dex-server")
+			}
+			verifyDexConnector(m)
 
-			connectors, ok := m["connectors"]
-			if !ok {
-				t.Fatal("no connectors found in dex.config")
+			// update .dex.config and verify that the dex connector is not overwritten
+			type expiry struct {
+				IdTokens    string `yaml:"idTokens"`
+				SigningKeys string `yaml:"signingKeys"`
 			}
-			dexConnector := connectors.([]interface{})[0].(map[interface{}]interface{})
-			config := dexConnector["config"]
-			assert.Equal(t, config.(map[interface{}]interface{})["clientID"], "system:serviceaccount:argocd:argocd-argocd-dex-server")
+
+			dexCfg := map[string]interface{}{
+				"expiry": expiry{
+					IdTokens:    "1hr",
+					SigningKeys: "12hr",
+				},
+			}
+
+			dexCfgBytes, err := yaml.Marshal(dexCfg)
+			assert.NoError(t, err)
+
+			a.Spec.SSO.Dex.Config = string(dexCfgBytes)
+
+			err = r.reconcileArgoConfigMap(a)
+			assert.NoError(t, err)
+
+			m = getDexConfig(t, r.Client)
+			_, ok := m["expiry"]
+			assert.True(t, ok)
+
+			verifyDexConnector(m)
```

Comment:
For your consideration, just a suggestion as you are more familiar with the test behaviour from working on it here:
- Would this part of the test work/make sense as it's own entry in the test struct (line ~427), rather than as an addition to the end of the test?
- e.g. in here:
```go
tests := []struct {
		name             string
		updateCrSpecFunc func(cr *argoproj.ArgoCD)
	}{
		{
```

Reply:
added as a separate test entry